### PR TITLE
Hook/add price table component

### PR DESCRIPTION
**Pull request by kelseythejackson**
## Description
We had a pattern we were using to show prices, so I pulled it out into it's own component.

We were also having trouble with buttons in outlook so I hacked together a custom button component to account for that.

## Before you ask for review

- [x] I've reviewed my own PR
- [x] I've made sure this branch is current with the base branch
- [x] I've added comments to the PR that will help reviewers

### What I'm looking for

@frrrances @robmadole I need a sanity check that on this. It works, but I want to make sure I'm not missing anything as far as: 

- test cases
- code quality

### Tasks for reviewers

- [x] Checkout this branch
- [x] Run `npm install` then `npm test` to see if everything passes.

### CODE REVIEW

File: test/pricingTable.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          <td>1</td>
+        </tr>
+        <tr>
+          <td>2</td>
+        </tr>
+      </table>
+    `;
+    compare(input, expected);
+  });
+});
```
Comment (kelseythejackson) [inline test/pricingTable.js:43]: This is probably the easiest way to show this. The `pricing-table` component can be used two ways,
either with simple title and total attrbutes:
`<price-table title="Something" total="$80"></price-table>`

Or with a heading and body for more flexibility:
`<price-table>
        <price-heading><td>1</td></price-heading>
        <price-body><td>2</td></price-body>
      </price-table>`

File: lib/componentFactory.js [hunk 2]
```diff
@@ -1,171 +1,364 @@
...
+      if (element.attr("href")) {
+        inner = format(
+          '<a %s href="%s"%s>%s</a>',
+          attrs,
+          element.attr("href"),
+          target,
+          inner
+        );
+      }
+
```
Comment (kelseythejackson) [inline lib/componentFactory.js]: The start of the button code

Comment (kelseythejackson) [reply lib/componentFactory.js]: Screens of the same email, Outlook and Apple:

Screen Shot 2022-06-02 at 9 24 49 AM
Screen Shot 2022-06-02 at 9 24 37 AM

Comment (frrrances) [reply lib/componentFactory.js]: 🎉

File: lib/componentFactory.js [hunk 3]
```diff
@@ -241,6 +241,122 @@ module.exports = function (element) {
...
+      // The .button class is always there, along with any others on the <button> element
+      var classes = ["button"];
+      if (element.attr("class")) {
+        classes = classes.concat(element.attr("class").split(" "));
+      }
+
+      return format(
+        `<!--[if mso]><div><v:roundrect xmlns:v="urn:schemas-microsoft-com:vml"
+    xmlns:w="urn:schemas-microsoft-com:office:word"
+    href="https://fontawesome.com/icons"
```
Comment (frrrances) [inline lib/componentFactory.js]: Looks like some values here are hard-coded (URL, height, width, and button text especially) that we may want to have as variables?

File: package.json [hunk 4]
```diff
@@ -24,14 +24,15 @@
...
     "assert-html-equal": "^1.0.0",
-    "babel-preset-es2015": "^6.24.1",
-    "babel-register": "^6.26.0",
-    "browserify": "^17.0.0",
-    "chalk": "^5.0.1",
-    "gulp": "^4.0.2",
-    "gulp-uglify": "^3.0.2",
-    "mocha": "^10.0.0",
-    "rimraf": "^3.0.2",
+    "babel-preset-es2015": "^6.3.13",
```
Comment (frrrances) [inline package.json:27]: Looks like some of these were moved down to earlier versions... was that intentional?

**Review by frrrances — APPROVED** [review]
Nice work @kelseythejackson! I left one comment about adjusting some hard-coded values into variables, but otherwise these changes make sense to me from a front-end perspective. I'll let @robmadole weigh in on the more technical side.

**Comment by robmadole** [comment]
@kelseythejackson this looks great. Frances caught the same questions I have too.

And one other. Now that we are adding component and tests can we get these running in GitHub Actions? Here's [an example of our test in fontawesome](https://github.com/FortAwesome/fontawesome/blob/master/.github/workflows/tests.yml). You might start with something like:

```yml
name: Inky Tests

on: [push]

jobs:
  frontend:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v1
      with:
        node-version: '12'
    - name: Install Dependencies
      run: |
        npm install -g npm@8.x
        npm install
    - name: Run Tests
      run: |
        npm run test
```

**Comment by robmadole** [comment]
@kelseythejackson this looks great. Frances caught the same questions I have too.

And one other. Now that we are adding component and tests can we get these running in GitHub Actions? Here's [an example of our test in fontawesome](https://github.com/FortAwesome/fontawesome/blob/master/.github/workflows/tests.yml). You might start with something like:

```yml
name: Inky Tests

on: [push]

jobs:
  frontend:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v1
      with:
        node-version: '12'
    - name: Install Dependencies
      run: |
        npm install -g npm@8.x
        npm install
    - name: Run Tests
      run: |
        npm run test
```

**Review by robmadole — APPROVED** [review]

**Pull request merged**